plugins/s3DropFolderPlugin/lib/model/S3DropFolderFile.php

```diff
@@ -21,6 +21,14 @@ public function getFileUrl()
 	{
 		$dbDropFolder = DropFolderPeer::retrieveByPK($this->getDropFolderId());
 		$s3Options = $dbDropFolder->getDropFolderParams();
+		
+		if ($s3Options['useS3Arn'] && empty($s3Options['s3Arn']))

```

Comment:
Same comment as above:
Why no fail it in the step of the drop folder creation? Would be much more clean.
In validate for insert